OpenRA.Mods.Common/Traits/BotModules/BaseBuilderBotModule.cs

```diff
@@ -220,14 +220,15 @@ void IBotTick.BotTick(IBot bot)
 			// PERF: We tick only one type of valid queue at a time
 			// if AI gets enough cash, it can fill all of its queues with enough ticks
 			var findQueue = false;
+			var queuesByCategory = AIUtils.FindQueuesByCategory(player);
 			for (int i = 0, builderIndex = currentBuilderIndex; i < builders.Length; i++)
 			{
 				if (++builderIndex >= builders.Length)
 					builderIndex = 0;
 
 				--builders[builderIndex].WaitTicks;
 
-				var queues = AIUtils.FindQueues(player, builders[builderIndex].Category).ToArray();
+				var queues = queuesByCategory[builders[builderIndex].Category].ToArray();
 				if (queues.Length != 0)
 				{
 					if (!findQueue)
```

Comment:
I noticed `findQueue` is set but never used i.e. to escape the for loop. Already in the past code.

Reply:
Perhaps the for loop is meant to iterate all builders to decrease `WaitTicks`. But then still why do `FindQueues` if `findQueue` is already true.